_posts/2025-03-28-this-week-in-rails.markdown

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: "Capturing error reports in tests and more"
+categories: news
+author: Wojtek
+og_image: assets/images/this-week-in-rails.png
+published: true
+date: 2025-03-28
+---
+
+Hi, [Wojtek](https://x.com/morgoth85) here. Let's explore last 2 week's changes in the Rails codebase.
+
+[Add capture_error_reports](https://github.com/rails/rails/pull/54809)  
+Captures all reported errors from within the block that match the given error class.
+
+```ruby
+reports = capture_error_reports(IOError) do
+  Rails.error.report(IOError.new("What a Terrible Failure"))
+  Rails.error.report(StandardError.new)
+end
+assert_equal 1, reports.size
+assert_equal "What a Terrible Failure", reports.first.error.message
+```
+
+[Don't always escape JSON when rendering from controller](https://github.com/rails/rails/pull/54643)  
+The JSON renderer doesn't escape HTML entities or Unicode line separators anymore.
+Since the renderer is used to return a JSON document as *application/json*, it's typically not necessary to escape those characters, and it improves performance.
+
+You can use the *escape: true* option or set *config.action_controller.escape_json_responses = true* to restore the escaping behavior.
+
+[Load lazy route sets before inserting test routes](https://github.com/rails/rails/pull/54777)  
+Without loading lazy route sets early, we missed *after_routes_loaded* callbacks, and risked invoking them with the test routes instead of the real ones if another load was triggered by an engine.
+
+[Add RuboCop cache to GHA workflow templates](https://github.com/rails/rails/pull/54754)  
+The aim is to speed up RuboCop checks by sharing the cache between workflow runs, when possible.
+
+[Guides update for PostgreSQL timestamp with time zone](https://github.com/rails/rails/pull/53687)  
+Document how to use a configuration option to change the default timestamp data type from *timestamp without time zone* to *timestamp with time zone*.
+  
```

Comment:
https://github.com/rails/rails/pull/54722 seems significant, worth adding?

Reply:
Thanks @vipulnsward. I added it. Since there was no changelog and looked like a deep internal change, I skipped it, but after a second look, it indeed may interest someone.